### PR TITLE
[#1586] improvement(netty): Allow Netty Worker thread pool size to dynamically adapt to the number of processor cores

### PR DESCRIPTION
**Pull request by rickyma**
### What changes were proposed in this pull request?

Allow Netty Worker thread pool size to dynamically adapt to the number of processor cores.
We will have a more reasonable default value and better performance in most cases.

### Why are the changes needed?

For https://github.com/apache/incubator-uniffle/issues/1586.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing UTs.

### CODE REVIEW

File: server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java [hunk 1]
```diff
@@ -500,8 +500,11 @@ public class ShuffleServerConf extends RssBaseConf {
   public static final ConfigOption<Integer> NETTY_SERVER_WORKER_THREAD =
       ConfigOptions.key("rss.server.netty.worker.thread")
           .intType()
-          .defaultValue(100)
-          .withDescription("Worker thread count in netty");
+          .defaultValue(0)
```
Comment (jerqi) [inline server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java:503]: Could you refer to the Spark similar implement? Does Spark use `0` or nodefault or other value?

Comment (rickyma) [reply server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java:503]: > Could you refer to the Spark similar implement? Does Spark use `0`, `-1` or nodefault or other value?

In Spark, there isn't a standard for default configuration values; some use `nil`, some use `0`. I think using 0 in our case is similar to the scenario of configuring the `spark.blockManager.port`:
The default value of `spark.blockManager.port` is set to 0, which indicates a random free port.

Comment (jerqi) [reply server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java:503]: You should refer to the thread number of Spark.

Comment (rickyma) [reply server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java:503]: > You should refer to the thread number of Spark.

Refer to `org.apache.spark.network.util.TransportConf`, the default value is also `0`
(In https://github.com/apache/spark/blob/master/common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java#L130):
```
public class TransportConf {
  ...
  /** Number of threads used in the server thread pool. Default to 0, which is 2x#cores. */
  public int serverThreads() { return conf.getInt(SPARK_NETWORK_IO_SERVERTHREADS_KEY, 0); }
  ...
}
```

**Review by jerqi — APPROVED** [review]
LGTM, thanks @rickyma

**Pull request merged**